tools/hidl/hidlcomp.cpp

```diff
@@ -4278,7 +4281,11 @@ void EspMessageInfo::write_esp()
     //method ==> copy from interface
     outf("\nvoid C%s::copy(IConst%s &ifrom)\n{\n", name_, name_);
     if (parent)
-        outf("\tC%s::copy(*dynamic_cast<IConst%s*>(&ifrom));\n", parent, parent);
+    {
+        outf(1, "C%s *classFrom = static_cast<C%s*>(&ifrom);\n", name_, name_);
+        outf(1, "IConst%s *baseICFrom = static_cast<IConst%s*>(classFrom);\n", parent, parent);
```

Comment:
@ghalliday a single static_cast wouldn't work but I think two static_casts is still much faster that dynamic_cast.